### PR TITLE
GenericActionDialog huge refactor

### PR DESCRIPTION
**Pull request by JulianWielga**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/client/.eslintignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 node_modules/
 src/brace/
-test/
+/test/
```
Comment (Dzuming) [inline designer/client/.eslintignore:3]: Why do we need this change?

Comment (JulianWielga) [reply designer/client/.eslintignore:3]: intention of this line was to filter out `client/test` but not `client/src/test`. this mistake costed us lots of poor lines in `src/test`

File: designer/client/src/containers/theme/nuTheme.tsx [hunk 2]
```diff
@@ -41,16 +70,35 @@ const custom = {
     fontSize: 14,
 };
 
-export const getDesignTokens = (mode: PaletteMode) => {
-    const modePalette: ModePalette = mode === "light" ? lightModePalette : darkModePalette;
+const extendWithHelpers = (custom: CustomPalette) => ({
```
Comment (Dzuming) [inline designer/client/src/containers/theme/nuTheme.tsx:73]: Nice

**Review by Dzuming — APPROVED** [review]
I added a few comments but they are non-blocking

File: designer/client/src/components/graph/node-modal/NodeAdditionalInfoBox.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React, { useCallback, useEffect, useState } from "react";
 import HttpService from "../../../http/HttpService";
-import ReactMarkdown from "react-markdown/with-html";
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply designer/client/src/components/graph/node-modal/NodeAdditionalInfoBox.tsx:3]: markdown-to-jsx supports custom components in really easy way.
also it's used in other places - wanted to reduce different libraries usages.

File: designer/client/src/components/graph/node-modal/nodeDetails/Subtype.tsx [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
-import { MODAL_HEADER_HEIGHT } from "../../../../stylesheets/variables";
+import { styled } from "@mui/material";
+import { IconModalTitle } from "./IconModalTitle";
+import { blendDarken, blendLighten } from "../../../../containers/theme/helpers";
+import { getLuminance } from "@mui/system/colorManipulator";
 
-import { blendLighten } from "../../../../containers/theme/helpers";
+export const getColorBlend = (color: string, value: number) =>
+    getLuminance(color) > 0.5 ? blendDarken(color, value) : blendLighten(color, value);
+export const getColorBlend2 = (color: string, value: number) =>
```
[the remark this answers is not in the document]
Comment (Dzuming) [reply designer/client/src/components/graph/node-modal/nodeDetails/Subtype.tsx:8]: It sound good to me

**Review by Dzuming — APPROVED** [review]

**Pull request merged**